ssz_serialization/merkleization.nim

```diff
@@ -520,8 +520,7 @@ func bitListHashTreeRoot(merkleizer: var SszMerkleizerImpl, x: BitSeq): Digest =
   if lastCorrectedByte == byte(1):
     if totalBytes == 1:
       # This is an empty bit list.
-      # It should be hashed as a tree containing all zeros:
-      return zeroHashes[merkleizer.topIndex]
+      return getFinalHash(merkleizer)
```

Comment:
What's the rationale here?

Reply:
Consistency with `chunkedHashTreeRoot`, and keeping the internal knowledge about how the merkleizer works inside the functions operating on just the merkleizer (less code duplication). 

`getFinalHash` starts with this optimization, so performance characteristics should not change.
```
func getFinalHash*(merkleizer: SszMerkleizerImpl): Eth2Digest =
  if merkleizer.totalChunks == 0:
    return zeroHashes[merkleizer.topIndex]
  # ...
```